packages/components/src/components/drawer/shadow.tsx

```diff
@@ -0,0 +1,177 @@
+		return (
+			<div
+				class={`drawer__wrapper drawer__wrapper--${this.state._align as string} ${this.state._open ? 'drawer__wrapper--open' : `drawer__wrapper--${this.state._align as string}--close`}`}
+				aria-label={this.state._label}
+			>
+				<div class="drawer__content">
+					<slot />
+				</div>
+			</div>
+		);
+	}
+
+	private getRef = (el: HTMLDialogElement | undefined) => (this.dialogElement = el as HTMLDialogElement);
+	public render(): JSX.Element {
+		const isModal = this.state._modal;
+		const isOpen = this.state._open;
+		return (
+			<Host class={`kol-drawer drawer ${isModal ? 'drawer--modal' : ''} ${isOpen ? 'drawer--open' : ''}`} ref={(el) => (this.hostElement = el as HTMLElement)}>
+				<dialog
+					class={`drawer__dialog drawer__dialog--${this.state._align as string} ${isOpen ? 'drawer__dialog--open' : `drawer__dialog--${this.state._align as string}--close`}`}
+					ref={this.getRef}
+				>
+					{this.renderDialogContent()}
+				</dialog>
+			</Host>
+		);
+	}
+
+	/**
+	 * Specifies the default open state of the drawer.
+	 */
+	@Prop() public _open?: OpenPropType;
+
+	/**
+	 * Specifies the orientation of the drawer.
+	 */
+	@Prop() public _align?: AlignPropType;
+
+	/**
+	 * Defines the visible or semantic label of the component (e.g. aria-label, label, headline, caption, summary, etc.).
+	 */
+	@Prop() public _label!: LabelPropType;
+
+	/**
+	 * Indicates whether the drawer is a modal.
+	 */
+	@Prop() public _modal?: ModalPropType;
+
+	/**
+	 * Specifies the EventCallback function to be called when the drawer is closing.
+	 */
+	@Prop() public _on?: KoliBriModalEventCallbacks;
+
+	@State() public state: DrawerStates = {
+		_label: '', // ⚠ required
+		_open: false,
+		_align: 'left',
+	};
+
+	@Watch('_label')
+	public validateLabel(value?: LabelPropType): void {
+		validateLabel(this, value, {
+			required: true,
+		});
+	}
+
+	@Watch('_align')
+	public validateAlign(value?: AlignPropType): void {
+		validateAlign(this, value);
+	}
+
+	@Watch('_modal')
+	public validateModal(value?: ModalPropType): void {
+		validateModal(this, value);
+	}
+
+	@Watch('_open')
+	public validateOpen(value?: OpenPropType): void {
+		if (typeof value === 'boolean') {
+			validateOpen(this, value);
+			value ? this.open() : this.close();
+		}
+	}
+
+	@Watch('_on')
+	public validateOn(value?: KoliBriModalEventCallbacks): void {
+		if (typeof value === 'object' && value !== null) {
+			const callbacks: KoliBriModalEventCallbacks = {};
+			if (typeof value.onClose === 'function') {
+				callbacks.onClose = value.onClose;
+			}
+			setState<KoliBriModalEventCallbacks>(this, '_on', callbacks);
+		}
+	}
+
+	private handleClose() {
+		this.close();
+		this.dialogElement?.close();
+		this._on?.onClose?.();
+	}
+
+	private handleAnimationEnd(e: Event) {
+		const animationEvent = e as AnimationEvent;
+		if (animationEvent.animationName.includes('slideOut')) {
+			this.dialogElement?.close();
+			this._on?.onClose?.();
```

Comment:
Hier fände ich es gut, wenn wir klar machen könnten, dass es das selbe Code-Fragment wie in L45 ist. Z.B. durch eine neue Methode. Sonst passt man zukünftig ggf. nur eine der beiden Stellen an.